gossip-base/src/main/java/org/apache/gossip/crdt/PNCounter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gossip.crdt;
+
+import org.apache.gossip.manager.GossipManager;
+
+public class PNCounter implements CrdtCounter<Long, PNCounter> {
+
+	private final GossipManager myManager;
+	private GrowOnlyCounter pCount;
+	private GrowOnlyCounter nCount;
+	
+	public PNCounter(GossipManager manager) {
```

Comment:
These objects need to be serializable/deserializable via Jackson. There is way to do construction injection with Jackson. However you are going to need other constructors. See https://github.com/apache/incubator-gossip/blob/master/gossip-base/src/main/java/org/apache/gossip/crdt/GrowOnlyCounter.java